### PR TITLE
Handle login by auth managers

### LINKED ISSUE #32192: AIP-56 - FAB AM - Login
Create the whole login process:

- Move the login form view to auth manager
- Move the login logic to auth manager
- Return the login URL as part of get_url_login() API

### PR DESCRIPTION
**Pull request by vincbeck**
Move the login logic to auth managers.

Resolves #32192

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code changes, an Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvement+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in a newsfragment file, named `{pr_number}.significant.rst` or `{issue_number}.significant.rst`, in [newsfragments](https://github.com/apache/airflow/tree/main/newsfragments).

### CODE REVIEW

File: airflow/auth/managers/fab/auth/anonymous_user.py [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+    def roles(self, roles):
+        self._roles = roles
+        self._perms = set()
+
+    @property
+    def perms(self):
+        if not self._perms:
+            self._perms = set()
+            for role in self.roles:
+                self._perms.update({(perm.action.name, perm.resource.name) for perm in role.permissions})
```
Comment (uranusjr) [inline airflow/auth/managers/fab/auth/anonymous_user.py]:
```suggestion
            self._perms = {
                (perm.action.name, perm.resource.name)
                for role in self.roles
                for perm in role.permissions
            }
```

File: airflow/auth/managers/fab/fab_auth_manager.py [hunk 2]
```diff
@@ -43,3 +46,16 @@ def get_user_name(self) -> str:
...
+    def get_security_manager_override_class(self) -> type:
+        """Return the security manager override."""
+        return FabAirflowSecurityManagerOverride
+
+    def get_url_login(self, **kwargs) -> str:
+        """Return the login page url."""
+        if not self.security_manager.auth_view:
+            raise AirflowException("`auth_view` not defined in the security manager.")
+        if "next_url" in kwargs and kwargs["next_url"]:
+            return url_for(f"{self.security_manager.auth_view.endpoint}.{'login'}", next=kwargs["next_url"])
```
Comment (uranusjr) [inline airflow/auth/managers/fab/fab_auth_manager.py]:
```suggestion
            return url_for(f"{self.security_manager.auth_view.endpoint}.login", next=kwargs["next_url"])
```

**Review by uranusjr — APPROVED** [review]
Looks good, just a couple to nitpicks that don’t affect the logic at all.

**Comment by vincbeck** [comment]
> #32525

Thanks @uranusjr for the review! If you have few seconds/minutes, could you review #32525? You added some comments there that I addressed. Thanks :)

**Review by o-nikolas — APPROVED** [review]

**Pull request merged**